server/src/main/java/org/candlepin/controller/ProductManager.java

```diff
@@ -832,6 +865,20 @@ public static boolean isChangedBy(Product entity, ProductDTO dto) {
             }
         }
 
+        Collection<ProductDTO> providedProducts = dto.getProvidedProducts();
+
+        if (providedProducts != null) {
+
+            if (!Util.collectionsAreEqual(entity.getProvidedProducts().stream()
+                .map(Product::getId)
```

Comment:
I'm unsure comparing the id only here is enough, instead of a proper comparator for the whole product. For other complex fields such as ProductContent and Branding we're doing a full comparison. @Ceiu what do you think?

Reply:
It's a bit inconsistent, but in the two examples you noted those entities either have no reliable/referenceable natural ID at the user level and provide the referenced entity wholesale (ProductContent, Branding), or themselves are changed by the ProductManager (ProductContent).

I think in this particular case since we're only ever caring about the reference to another product and never updating the referenced product here, and the reference itself carries no metadata like ProductContent does via "enabled," we should be good by only checking the [RH] ID. The question here, however, becomes one of which ID should we use? There's a fair argument for both, but I suppose using the RHID rather than the UUID prevents us from breaking org boundries, so long as we're consistent in ensuring the RHID actually exists in the org before assignment.